protocol/contracts/templegold/StakedTempleVoteToken.sol

```diff
@@ -0,0 +1,204 @@
+// Temple (templegold/StakedTempleVoteToken.sol)
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20Burnable } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import { Pausable } from "@openzeppelin/contracts/utils/Pausable.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { TempleElevatedAccess } from "contracts/v2/access/TempleElevatedAccess.sol";
+import { IStakedTempleVoteToken } from "contracts/interfaces/templegold/IStakedTempleVoteToken.sol";
+import { ITempleGoldStaking } from "contracts/interfaces/templegold/ITempleGoldStaking.sol";
+
+/** 
+ * @title Staked Temple Vote Token
+ * @notice Non-transferrable ERC20 token issued by Temple Gold Staking contract. 
+ * StakedTempleVoteToken gives governance power to holders and delegates by enabling voting
+ */
+contract StakedTempleVoteToken is IStakedTempleVoteToken, TempleElevatedAccess, ERC20, ERC20Burnable, Pausable {
+
+    /// @notice Staking contract. Mutable if ever staking contract is upgraded.
+    address public override staking;
+    /// @notice Authorized contracts that can burn/mint
+    mapping(address authority => bool authorized) public override authorized;
+    
+    constructor(
+        address _initialRescuer,
+        address _initialExecutor,
+        address _staking,
+        string memory _name,
+        string memory _symbol
+    ) ERC20(_name, _symbol) TempleElevatedAccess(_initialRescuer, _initialExecutor) {
+        staking = _staking;
+    }
+
+    /**  
+     * @notice Set staking contract
+     * @param _staking Staking contract
+     */
+    function setStaking(address _staking) external override onlyElevatedAccess {
+        if (_staking == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        staking = _staking;
+        emit StakingSet(_staking);
+    }
+
+    /**  
+     * @notice Set authorized contract
+     * @dev Aucthorized contract is allowed to transfer/burn vote token
+     * @param _authority Contract
+     * @param _authorized Bool if authorized
+     */
+    function setAuthorized(address _authority, bool _authorized) external override onlyElevatedAccess {
+        if (_authority == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        authorized[_authority] = _authorized;
+        emit AuthoritySet(_authority, _authorized);
+    }
+
+    /**  
+     * @notice Mint vote token
+     * @dev Only authorized can mint
+     * @param to Recipient
+     * @param amount Amount to mint
+     */
+    function mint(address to, uint256 amount) external override onlyAuthorized whenNotPaused {
+        _mint(to, amount);
+    }
+
+    /// @notice override to stop holders burning vote tokens themselves
+    function burn(uint256 /*amount*/) public virtual override(IStakedTempleVoteToken, ERC20Burnable) {
```

Comment:
I think it would be better to NOT inherit the ERC20Burnable() contract in this case and just implement the one burnFrom() function yourself.